src/App.tsx

```diff
@@ -1,223 +1,111 @@
+import React, {
+  useCallback, useContext, useEffect, useMemo, useState,
+} from 'react';
+import { deleteTodoById, getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { CompletedFilter } from './types/CompletedFilter';
+import { Header } from './components/Header';
+import { Todolist } from './components/Todolist';
+import { Footer } from './components/Footer';
+import { Errornotification } from './components/Errornotification';
+import { filteredTodos } from './helper/Helper';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [completedFilter, setCompletedFilter] = useState(CompletedFilter.All);
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo | null>(null);
+  const [deletingTodosIds, setDeletingTodosIds] = useState([0]);
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const showError = useCallback((message: string) => {
+    setErrorMessage(message);
+
+    setTimeout(() => setErrorMessage(''), 3000);
+  }, []);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => showError('Can\'t load todos'));
     }
   }, []);
 
+  const closeErrorMessage = useCallback(() => {
+    setErrorMessage('');
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    return filteredTodos(todos, completedFilter);
+  }, [todos, completedFilter]);
+
+  const handleDeleteTodo = (todoId: number) => {
+    setDeletingTodosIds(
+      (currentDeletetingTodos) => [...currentDeletetingTodos, todoId],
+    );
+
+    deleteTodoById(todoId)
+      .then(() => {
+        setTodos(currentTodos => currentTodos
+          .filter(currentTodo => currentTodo.id !== todoId));
+
+        setDeletingTodosIds(
+          (currentDeletetingTodos) => (
+            currentDeletetingTodos.filter(deletingId => deletingId !== todoId)
+          ),
+        );
+      })
+      .catch(() => showError('Unable to delete a todo'));
+  };
+
+  const removeCompletedTodos = () => {
+    todos.forEach(todo => {
+      if (todo.completed) {
+        handleDeleteTodo(todo.id || 0);
+      }
+    });
+  };
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
-        <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
+        <Header
+          setTemporaryTodo={setTemporaryTodo}
+          showError={showError}
+          setTodos={setTodos}
```

Comment:
I decided earlier to leave it in Header, because want to reduce the amount of code in App.
in the third part in App, I have 170 strings, 
the header has 97 strings with the method AddTodo. it's ok if I leave it here?